signal/signal.go

```diff
@@ -132,8 +132,12 @@ func (s *Signal) unblockShutdown(id int) {
 	numBlocking := len(s.blocking)
```

Comment:
I know it's outside of the scope of this PR, but I don't understand why `s.blocking` is a map instead of just an atomic counter. The shutdown process looks very complicated.

Reply:
Since it turned out not to be a big change I updated the code.